### ISSUE TITLE
Training Log for 2x Upscaling

### ISSUE THREAD

**Issue by alexjc** [alexjc/neural-enhance#63]
Running this command from `v0.3` branch gives this log on a TitanX under Ubuntu 14.4 and latest CUDA: 

```
python3.4 enhance.py \
    --train "$OPEN_IMAGES_PATH/*/*.jpg" --model medium \
    --epochs=50 --batch-shape=256 --device=gpu0 \
    --generator-downscale=1 --generator-upscale=2 \
    --generator-blocks=8 --generator-filters=128 --generator-residual=0 \
    --perceptual-layer=conv2_2 --smoothness-weight=1e7 --adversary-weight=0.0 \
    --train-blur=1 --train-noise=1.0
```

It's training on 1M images (not downscaled), but 50k should work fine too. The log is the following:

```
Epoch #1 at 209.2s, lr=1.00e-04
  - generator total=8.44e+04 prcpt=8.41e+04 smthn=3.07e+02
Epoch #2 at 417.4s, lr=1.00e-04
  - generator total=6.26e+04 prcpt=6.14e+04 smthn=1.16e+03
Epoch #3 at 625.8s, lr=1.00e-04
  - generator total=4.69e+04 prcpt=4.52e+04 smthn=1.74e+03
Epoch #4 at 834.5s, lr=1.00e-04
  - generator total=3.44e+04 prcpt=3.28e+04 smthn=1.57e+03
Epoch #5 at 1042.9s, lr=1.00e-04
  - generator total=3.64e+04 prcpt=3.44e+04 smthn=1.96e+03
Epoch #6 at 1251.5s, lr=1.00e-04
  - generator total=3.19e+04 prcpt=3.00e+04 smthn=1.89e+03
Epoch #7 at 1459.6s, lr=1.00e-04
  - generator total=3.16e+04 prcpt=2.96e+04 smthn=1.97e+03
Epoch #8 at 1668.5s, lr=1.00e-04
  - generator total=3.00e+04 prcpt=2.78e+04 smthn=2.24e+03
Epoch #9 at 1877.1s, lr=1.00e-04
  - generator total=3.24e+04 prcpt=2.99e+04 smthn=2.52e+03
Epoch #10 at 2085.9s, lr=1.00e-04
  - generator total=3.17e+04 prcpt=2.89e+04 smthn=2.76e+03
Epoch #11 at 2295.8s, lr=1.00e-04
  - generator total=3.46e+04 prcpt=3.18e+04 smthn=2.72e+03
Epoch #12 at 2504.6s, lr=1.00e-04
  - generator total=3.24e+04 prcpt=2.99e+04 smthn=2.45e+03
Epoch #13 at 2713.4s, lr=1.00e-04
  - generator total=2.65e+04 prcpt=2.43e+04 smthn=2.22e+03
Epoch #14 at 2922.1s, lr=1.00e-04
  - generator total=2.33e+04 prcpt=2.11e+04 smthn=2.11e+03
Epoch #15 at 3131.5s, lr=1.00e-04
  - generator total=2.49e+04 prcpt=2.26e+04 smthn=2.31e+03
Epoch #16 at 3340.3s, lr=1.00e-04
  - generator total=2.68e+04 prcpt=2.42e+04 smthn=2.62e+03
Epoch #17 at 3549.1s, lr=1.00e-04
  - generator total=2.38e+04 prcpt=2.14e+04 smthn=2.39e+03
Epoch #18 at 3758.0s, lr=1.00e-04
  - generator total=2.42e+04 prcpt=2.20e+04 smthn=2.20e+03
Epoch #19 at 3967.4s, lr=1.00e-04
  - generator total=2.33e+04 prcpt=2.12e+04 smthn=2.13e+03
Epoch #20 at 4176.8s, lr=1.00e-04
  - generator total=2.31e+04 prcpt=2.06e+04 smthn=2.47e+03
Epoch #21 at 4386.7s, lr=1.00e-04
  - generator total=2.67e+04 prcpt=2.40e+04 smthn=2.73e+03
Epoch #22 at 4595.9s, lr=1.00e-04
  - generator total=2.31e+04 prcpt=2.07e+04 smthn=2.39e+03
Epoch #23 at 4804.8s, lr=1.00e-04
  - generator total=2.07e+04 prcpt=1.82e+04 smthn=2.50e+03
  Epoch #24 at 5014.1s, lr=1.00e-04
  - generator total=2.53e+04 prcpt=2.28e+04 smthn=2.57e+03
Epoch #25 at 5223.1s, lr=1.00e-04
  - generator total=1.97e+04 prcpt=1.77e+04 smthn=2.05e+03
Epoch #26 at 5431.7s, lr=1.00e-04
  - generator total=1.98e+04 prcpt=1.79e+04 smthn=1.96e+03
Epoch #27 at 5640.7s, lr=1.00e-04
  - generator total=2.07e+04 prcpt=1.84e+04 smthn=2.29e+03
Epoch #28 at 5849.8s, lr=1.00e-04
  - generator total=1.73e+04 prcpt=1.51e+04 smthn=2.13e+03
Epoch #29 at 6059.0s, lr=1.00e-04
  - generator total=1.78e+04 prcpt=1.57e+04 smthn=2.02e+03
Epoch #30 at 6267.9s, lr=1.00e-04
  - generator total=2.07e+04 prcpt=1.87e+04 smthn=2.08e+03
```

**Comment by sunsided** [comment]
Twitter guy here, thanks again! No learning rate adjustments at all? Or maybe the question is: If it were to learn longer, would there be noticeable improvements?

I'm actually constructing a network regressing 2D coordinates from images and try to learn from your architecture, hence the possibly stupid questions.

**Comment by alexjc** [comment]
@sunsided Hi, thanks for the questions! This is pre-training, and it only does 50 epochs. You see the results after 5 already, even the first epoch has visible shapes with brown colors.

The longer you train the better it is.  With the GAN, decreasing the learning rate is a bit harder to do as both NN are adapting to each other and it's not clear when they "converge" and when to decrease the rate. Doing it every 75 epochs is my current approach.

**Comment by eeric** [comment]
--python3 enhance.py --train 'images/*.jpg' --model custom --scales=2 --epochs=50 --perceptual-layer=conv2_2 --smoothness-weight=1e7 --adversary-weight=0.0 --generator-blocks=4 --generator-filters=64

WARNING: Could not load `/home/yq/work/Super-Resolution-using-Generative-Adversarial-Networks-master/srtrain_32/images/BAIDU_1735.jpg` as image.
  - Try fixing or removing the file before next run.
how to modify?

**Comment by alexjc** [comment]
@eeric Likely the image is too small for the default batch size. Remove it?

**Comment by eeric** [comment]
ok
